app/src/main/java/com/fstyle/structure_android/widget/dialog/MainAlertDialog.java

```diff
@@ -0,0 +1,191 @@
+import android.widget.TextView;
+import com.fstyle.structure_android.R;
+
+/**
+ * Created by le.quang.dao on 15/03/2017.
+ */
+
+public class MainAlertDialog extends AppCompatDialog implements DialogInterface {
+
+    @StyleRes
+    private int mTheme;
+    private AlertParams mParams;
+
+    MainAlertDialog(AlertParams alertParams, @StyleRes int theme) {
+        super(alertParams.mContext);
+        mParams = alertParams;
+        mTheme = theme;
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        Window window = getWindow();
+        if (window != null) {
+            // dim the screen to match the design
+            WindowManager.LayoutParams windowParams = window.getAttributes();
+            windowParams.dimAmount = 0.85f;
+            windowParams.flags = WindowManager.LayoutParams.FLAG_DIM_BEHIND;
+            window.setAttributes(windowParams);
+        }
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Window window = getWindow();
+        if (window != null) {
+            setTheme(mTheme);
+            window.getAttributes().width = ViewGroup.LayoutParams.MATCH_PARENT;
+            window.getAttributes().height = ViewGroup.LayoutParams.WRAP_CONTENT;
+            window.setBackgroundDrawable(new ColorDrawable(android.graphics.Color.TRANSPARENT));
+        }
+
+        TextView mTextViewTitle = (TextView) findViewById(R.id.tvTitle);
+        TextView mTextViewMessage = (TextView) findViewById(R.id.tvMessage);
+        Button mButtonNegative = (Button) findViewById(R.id.btn_negative);
+        Button mButtonPositive = (Button) findViewById(R.id.btn_positive);
+
+        mTextViewTitle.setVisibility(mParams.mInvisibleTitle ? View.GONE : View.VISIBLE);
+        mTextViewTitle.setText(mParams.mTitle);
+        mTextViewMessage.setText(mParams.mMessage);
+        assert mButtonNegative != null;
+        mButtonNegative.setVisibility(mParams.mInvisibleButtonCancle ? View.GONE : View.VISIBLE);
+        mButtonNegative.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                assert mParams.mNegativeListener != null;
+                mParams.mNegativeListener.onClick(MainAlertDialog.this,
+                        DialogInterface.BUTTON_NEGATIVE);
+            }
+        });
+        assert mButtonPositive != null;
+        mButtonPositive.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                assert mParams.mPositiveListener != null;
+                mParams.mPositiveListener.onClick(MainAlertDialog.this,
+                        DialogInterface.BUTTON_POSITIVE);
+            }
+        });
+    }
+
+    void setTheme(int theme) {
+        final TypedArray a =
+                getContext().obtainStyledAttributes(theme, R.styleable.MainAlertDialog);
```

Comment:
WARNING: [Recycle] This `TypedArray` should be recycled after use with `#recycle()`
 Many resources, such as TypedArrays, VelocityTrackers, etc., should be recycled (with a `recycle()` call) after use. This lint check looks for missing `recycle()` calls.